### PR TITLE
Feature/39519

### PR DESCRIPTION
**Pull request by sa11616079**
Installed following packages as dev dependencies:
- @babel/core
- @babel/node
- @babel/preset-env

Created .babelrc file with following content

~~~
{
"presets": ["@babel/preset-env"]
}
~~~

Add a file .gitignore in root directory and add node_modules and package-lock.json into it.

Created a file index.js inside extra folder.
Running the file index.js with babel and also write script for the same in package.json.

Command will be:
~~~
npx babel-node extra
~~~

Patterns:
Export the functions of the patterns from each file(using export default), Also removing function calling from these files.
Import all the patterns inside patterns/index.js and again export all the patterns from index.js file.
Import all the functions in extra/index.js exported from extra/patterns/index.js and now execute these functions with various arguments.

Utils:
-Permission:
* Create a file constants inside folder extra.
* Move all the constant values to above file.
* Export all constants using export.
* Remove hasPermission callings from permission.js.
* Export function hasPermission using export default.
* Import the hasPermission inside utils/index.js and export again using export.
* Import hasPermission inside extra/index.js and call the function with various arguments.

-Validation:
* Create a file helpers.js inside the utils folder.
* Move the function validateEmail from validation.js to helpers.js and export it using export.
* Import validateEmail from helpers.js inside validation.js.
* Export and import the funtion the same as done for hasPermission above.
* Move the constant users and function calling from validation.js to extra/index.js.

### CODE REVIEW

File: extra/pattern/index.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import diamond from './Diamond.js';
+import equilateral from './Equilateral.js';
```
Comment (SHUBHAM-JAIN-SS) [inline extra/pattern/index.js:2]: could use shorthand: export {default as module} from './module.js'

File: extra/utils/validation.js [hunk 2]
```diff
@@ -1,49 +1,28 @@
...
-    {
-    traineeEmail: "satish14.patel@successive.tech",
-    reviewerEmail: "shubham4@.jain@successive.tech"
-}
-];
 let checkemail = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
-let vldUser="",inVldUser="",count1=0,count2=0;
-function validateEmail(email)
+let vldUser = [], inVldUser = [];
+function validateEmail(email)
```
Comment (SHUBHAM-JAIN-SS) [inline extra/utils/validation.js:3]: this has to move to helpers

File: extra/utils/helpers.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+//helper.js
+
+const users =
```
Comment (SHUBHAM-JAIN-SS) [inline extra/utils/helpers.js:3]: should be in constants.js

**Review by SHUBHAM-JAIN-SS — CHANGES_REQUESTED** [review]
please make the required changes.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+node_modules
+package-lock.json
+.babelrc
```
Comment (SHUBHAM-JAIN-SS) [inline .gitignore:3]: should not be in gitignore

**Review by SHUBHAM-JAIN-SS — CHANGES_REQUESTED** [review]
please fix .gitignore file

**Comment by sa11616079** [comment]
> please fix .gitignore file

sir, I did't get what are you trying to explain me?

**Pull request merged**